src/provider/remoteFileSystemProvider.ts

```diff
@@ -310,24 +312,27 @@ export class VirtualFileSystem {
                 const doc = res.fileEntity as DocumentEntity;
                 if (update.v===doc.version) {
                     doc.version += 1;
-                    if (update.op && doc.cache) {
-                        let content = doc.cache;
+                    if (update.op && doc.remoteCache) {
+                        let content = doc.remoteCache;
                         update.op.forEach((op) => {
                             if (op.i) {
                                 content = content.slice(0, op.p) + op.i + content.slice(op.p);
                             } else if (op.d) {
                                 content = content.slice(0, op.p) + content.slice(op.p+op.d.length);
                             }
                         });
-                        doc.cache = content;
+                        if (! vscode.window.activeTextEditor?.document.isDirty){
```

Comment:
not active text editor, should be like: `res.path --> uri --> get text editor by uri`

Reply:
also note down why check it is dirty.